File: backend/app/schemas/analytics.py
import uuid
from datetime import date
from pydantic import BaseModel


class AnalyticsSummary(BaseModel):
    from_date: date
    to_date: date

    tasks_created: int
    tasks_completed: int

    habit_completions: int
    active_habits: int


class HabitHistory(BaseModel):
    habit_id: uuid.UUID
    from_date: date
    to_date: date
    completion_dates: list[date]

File: frontend/src/api/analytics.ts
import { api } from "./client";

export type AnalyticsSummary = {
  from_date: string;
  to_date: string;
  tasks_created: number;
  tasks_completed: number;
  habit_completions: number;
  active_habits: number;
};

export async function getSummary(from: string, to: string) {
  const res = await api.get<AnalyticsSummary>("/analytics/summary", {
    params: { from, to },
  });
  return res.data;
}